src/main/server.ts

```diff
@@ -1,17 +1,20 @@
 #!/usr/bin/env node
 
 import './ts-paths-bootstrap'
-import { cpus } from 'os'
 import * as config from 'config'
 import * as toBoolean from 'to-boolean'
 import * as express from 'express'
 import * as fs from 'fs'
 import * as path from 'path'
 import * as https from 'https'
-import * as cluster from 'cluster'
+
+import { app } from './app'
+import { ApplicationCluster } from './applicationCluster'
 
 const logger = require('@hmcts/nodejs-logging').getLogger('server')
 
+const defaultPort = 3000
```

Comment:
Default port should be close to the place it is needed, Please move it to the method that returns application port.

Other thing, since this is more complex shall we extract logic to separate class and make it something like `PortUtil` or `PortFinder`?